File: specification/specs2/indices/mapping_management/put_mapping/PutMappingRequest.ts
class PutMappingRequest extends RequestBase {
	all_field: AllField;
	date_detection: boolean;
	dynamic_date_formats: string[];
	dynamic_templates: Map<string, DynamicTemplate>;
	dynamic: Union<boolean, DynamicMapping>;
	field_names_field: FieldNamesField;
	index_field: IndexField;
	meta: Map<string, any>;
	numeric_detection: boolean;
	properties: Map<PropertyName, Property>;
	routing_field: RoutingField;
	size_field: SizeField;
	source_field: SourceField;
	@request_parameter()
	timeout: Time;
	@request_parameter()
	master_timeout: Time;
	@request_parameter()
	ignore_unavailable: boolean;
	@request_parameter()
	allow_no_indices: boolean;
	@request_parameter()
	expand_wildcards: ExpandWildcards;
	@request_parameter()
	update_all_types: boolean;
}

File: output/csharp/Indices/MappingManagement/PutMapping/PutMappingRequest.cs
using Nest.Common;
using Nest.CommonOptions;
using Nest.Mapping;
using Nest.CommonAbstractions;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Indices {

	public class PutMappingRequest : RequestBase {
		
		[DataMember(Name="allow_no_indices")]
		public bool AllowNoIndices { get; set; }


		[DataMember(Name="expand_wildcards")]
		public ExpandWildcards ExpandWildcards { get; set; }


		[DataMember(Name="ignore_unavailable")]
		public bool IgnoreUnavailable { get; set; }


		[DataMember(Name="include_type_name")]
		public bool IncludeTypeName { get; set; }


		[DataMember(Name="master_timeout")]
		public Time MasterTimeout { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="all_field")]
		public AllField AllField { get; set; }


		[DataMember(Name="date_detection")]
		public bool DateDetection { get; set; }


		[DataMember(Name="dynamic")]
		public Union<bool, DynamicMapping> Dynamic { get; set; }


		[DataMember(Name="dynamic_date_formats")]
		public List<string> DynamicDateFormats { get; set; }


		[DataMember(Name="dynamic_templates")]
		public IDictionary<string, DynamicTemplate> DynamicTemplates { get; set; }


		[DataMember(Name="field_names_field")]
		public FieldNamesField FieldNamesField { get; set; }


		[DataMember(Name="index_field")]
		public IndexField IndexField { get; set; }


		[DataMember(Name="meta")]
		public IDictionary<string, object> Meta { get; set; }


		[DataMember(Name="numeric_detection")]
		public bool NumericDetection { get; set; }


		[DataMember(Name="properties")]
		public IDictionary<PropertyName, IProperty> Properties { get; set; }


		[DataMember(Name="routing_field")]
		public RoutingField RoutingField { get; set; }


		[DataMember(Name="size_field")]
		public SizeField SizeField { get; set; }


		[DataMember(Name="source_field")]
		public SourceField SourceField { get; set; }

	}
}
